docs/source/routing/query-planning/query-planning-best-practices.mdx

```diff
@@ -165,6 +165,123 @@ generates two queries, one to each subgraph, like so:
 
 Currently, the router expands all interfaces into implementing types.
 
+## Troubleshooting query plans
+
+When investigating query plan behavior or performance issues, it's crucial to understand that query plans are generated based on multiple runtime and build-time factors. The best analogy for query planning is Google Maps: just as a route between two points is deterministic given the same inputs, query plans are deterministic when all factors remain constant.
+
+### Understanding query plan determinism
+
+Like Google Maps calculating the most efficient route from point A to point B, the Apollo Router determines the optimal path to resolve your GraphQL operation. The "route" remains consistent as long as the underlying conditions don't change. However, just as adding new roads, construction detours, changing speed limits, or current traffic patterns can alter your GPS route, changes to your federated graph can impact query planning decisions.
+
+The Apollo Router considers several inputs when generating query plans:
+
+- The GraphQL operation - The specific query, mutation, or subscription being executed
+- Supergraph schema - The composed schema from all your subgraphs
+- Query planner version - Tied to your specific router version
+- Router configuration - Including URL overrides, coprocessor logic, and other runtime config settings
+- Directive usage - How @shareable, @requires, progressive @override, @provides, and other directives are implemented across subgraphs.
```

Comment:
```suggestion
- Directive usage - How `@shareable`, `@requires`, progressive `@override`, `@provides`, and other directives are implemented across subgraphs.
```
Going to commit this small change so that AI Librarian runs and helps us out with the first pass review!